builtin/mainmenu/dlg_config_world.lua

```diff
@@ -54,137 +38,187 @@ local function get_formspec(data)
 		"button[3.25,7;2.5,0.5;btn_config_world_save;" .. fgettext("Save") .. "]" ..
 		"button[5.75,7;2.5,0.5;btn_config_world_cancel;" .. fgettext("Cancel") .. "]"
 
-	if mod ~= nil and mod.name ~= "" and mod.typ ~= "game_mod" then
+	if mod.name ~= "" and mod.typ ~= "game_mod" then
 		if mod.is_modpack then
 			local rawlist = data.list:get_raw_list()
 
 			local all_enabled = true
 			for j=1,#rawlist,1 do
 				if rawlist[j].modpack == mod.name and
-					rawlist[j].enabled ~= true then
+					not rawlist[j].enabled then
 						all_enabled = false
 						break
 				end
 			end
 
-			if all_enabled == false then
-				retval = retval .. "button[5.5,0.125;2.5,0.5;btn_mp_enable;" .. fgettext("Enable MP") .. "]"
-			else
+			if all_enabled then
 				retval = retval .. "button[5.5,0.125;2.5,0.5;btn_mp_disable;" .. fgettext("Disable MP") .. "]"
-			end
-		else
-			if mod.enabled then
-				retval = retval .. "checkbox[5.5,-0.125;cb_mod_enable;" .. fgettext("enabled") .. ";true]"
 			else
-				retval = retval .. "checkbox[5.5,-0.125;cb_mod_enable;" .. fgettext("enabled") .. ";false]"
+				retval = retval .. "button[5.5,0.125;2.5,0.5;btn_mp_enable;" .. fgettext("Enable MP") .. "]"
 			end
+		elseif mod.enabled then
+			retval = retval .. "checkbox[5.5,-0.125;cb_mod_enable;" .. fgettext("enabled") .. ";true]"
+		else
+			retval = retval .. "checkbox[5.5,-0.125;cb_mod_enable;" .. fgettext("enabled") .. ";false]"
```

Comment:
You simplified "Hide Game" and "Hide mp content" but not this enable checkbox - why?